admins/pageflow/user.rb

```diff
@@ -28,164 +23,273 @@ module Pageflow
+    collection_action 'delete_me',
+                      title: I18n.t('pageflow.admin.users.account'), method: [:get, :delete] do
       if request.delete?
         if current_user.destroy_with_password(params.require(:user)[:current_password])
-          redirect_to admin_root_path, :notice => I18n.t('pageflow.admin.users.me.updated')
+          redirect_to admin_root_path, notice: I18n.t('pageflow.admin.users.me.updated')
         end
       end
     end
 
-    member_action :resend_invitation, :method => :post do
+    member_action :resend_invitation, method: :post do
       InvitedUser.find(params[:id]).send_invitation!
-      redirect_to :back, :notice => I18n.t('pageflow.admin.users.resent_invitation')
+      redirect_to :back, notice: I18n.t('pageflow.admin.users.resent_invitation')
     end
 
-    member_action :suspend, :method => :post do
+    member_action :suspend, method: :post do
       User.find(params[:id]).suspend!
-      redirect_to :back, :notice => I18n.t('pageflow.admin.users.suspended')
+      redirect_to :back, notice: I18n.t('pageflow.admin.users.suspended')
     end
 
-    member_action :unsuspend, :method => :post do
+    member_action :unsuspend, method: :post do
       User.find(params[:id]).unsuspend!
-      redirect_to :back, :notice => I18n.t('pageflow.admin.users.unsuspended')
+      redirect_to :back, notice: I18n.t('pageflow.admin.users.unsuspended')
     end
 
     controller do
       include Pageflow::QuotaVerification
 
-      helper Pageflow::Admin::UsersHelper
-      helper Pageflow::Admin::LocalesHelper
       helper Pageflow::Admin::FormHelper
+      helper Pageflow::Admin::LocalesHelper
+      helper Pageflow::Admin::MembershipsHelper
       helper Pageflow::QuotaHelper
 
       def build_new_resource
-        user = InvitedUser.new(permitted_params[:user])
-        user.account ||= current_user.account
-        user
+        InvitedUser.new(permitted_params[:user])
       end
 
       def create_resource(user)
-        verify_quota!(:users, resource.account)
-        super
+        known_user = User.find_by(email: resource.email)
+        if known_user
+          membership_params = {user: known_user,
+                               entity_id: resource.initial_account,
+                               entity_type: 'Pageflow::Account'}
+          if resource.initial_role.present?
+            membership_params.merge!(role: resource.initial_role.to_sym)
+          end
+          Membership.create(membership_params)
+        else
+          verify_quota!(:users, params[:user][:account])
+          membership_params = {user: resource,
+                               entity_id: resource.initial_account,
+                               entity_type: 'Pageflow::Account'}
+          if resource.initial_role.present?
+            membership_params.merge!(role: resource.initial_role.to_sym)
+          end
+          Membership.create(membership_params)
```

Comment:
By restructuring the method to first only call `super` if user is not known, the duplication of membership creation code can be removed. Why does quota verification only happen for new users? Is there a test case missing?